pkg/v3/tickers/block.go

```diff
@@ -0,0 +1,74 @@
+package tickers
+
+import (
+	"context"
+	"log"
+	"sync"
+
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+)
+
+type blockSubscriber interface {
+	// Subscribe provides an identifier integer, a new channel, and potentially an error
+	Subscribe() (int, chan ocr2keepers.BlockHistory, error)
+	// Unsubscribe requires an identifier integer and indicates the provided channel should be closed
+	Unsubscribe(int) error
+}
+
+// blockTicker is a struct that follows the same design paradigm as a time ticker but provides blocks
+// instead of time
+type blockTicker struct {
+	C          chan ocr2keepers.BlockHistory
+	chID       int
+	ch         chan ocr2keepers.BlockHistory
+	subscriber blockSubscriber
+	closer     sync.Once
+	ctx        context.Context
+	cancel     context.CancelFunc
+}
+
+func NewBlockTicker(subscriber blockSubscriber) (*blockTicker, error) {
+	chID, ch, err := subscriber.Subscribe()
```

Comment:
I think it would make more sense to put the subscribe in the Start method. Then the `Once` functionality would not be needed on close.

Reply:
I did it this way so that the channel ID would be available to the instance as soon as client had created an instance with this constructor, since the Close() method passes the channel ID to the unsubscribe?

Reply:
Ahhh. That makes sense. It's the same challenge with the context.